### ISSUE TITLE
Rewrite `convert_object_elf_to_loadable_file` to use `object::elf` structs

### ISSUE THREAD

**Issue by bjorn3** [bytecodealliance/wasmtime#1856]
`object::elf` export type definitions for all important structs used here. You can use `object::pod::from_bytes` and `object::pos::bytes_of` to convert between a slice and these structs. A `from_bytes_mut` method would be useful for this though. You should be able to create one by adapting `from_bytes`.

_Originally posted by @bjorn3 in https://github.com/bytecodealliance/wasmtime/pull/1848_

**Comment by yurydelendik** [comment]
I created #1860 to partially address the issue. Though I had a little success with the object v0.19's pod API to refactor convert_object_elf_to_loadable_file. There are gaps in this API to make it simple.

**Comment by philipc** [comment]
@yurydelendik What is missing besides `from_bytes_mut`? Some of what you have done can be simplified already. For example:
```diff
-    let header: &FileHeader64<LittleEndian> =
-        unsafe { &*(bytes.as_mut_ptr() as *const FileHeader64<_>) };
+    let header = from_bytes::<FileHeader64<LittleEndian>>(&bytes).unwrap().0;
```

I thought about adding a `&mut [u8]` equivalent of `Bytes`, but I think it would be unusable with rust's borrowing rules.

**Comment by yurydelendik** [comment]
Yes, the immutable/reading part worked. Once I started patching, it all felt apart (including borrowing rules you mentioned).

**Comment by bjorn3** [comment]
I think this is fixed since #1860.
